### PR TITLE
all routing points

### PR DESCRIPTION
**Pull request by AristideEU**
I haven't been able to run the project, so take this with a grain of salt

### CODE REVIEW

File: src/extercom/mod.rs [hunk 1]
```diff
@@ -3,16 +3,98 @@ use rocket::{*, serde::json::Json};
 
 type Central = State<()>;
 
-/// Create a new file in the system. 
+type Id = usize;
+
+// Enum to describe speed
+#[derive(Serialize, Deserialize)]
+#[serde(crate = "rocket::serde")]
```
Comment (E-gy) [inline src/extercom/mod.rs:10]: No need strictly speaking, we've added `serde` to root dependencies anyway

File: src/extercom/mod.rs [hunk 2]
```diff
@@ -3,16 +3,98 @@ use rocket::{*, serde::json::Json};
...
+    High
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(crate = "rocket::serde")]
+struct Request {
+    uid:        Option<Id>,
+    pace:       Option<Pace>,
+    filename:   Option<String>,
+    text:       Option<String>
```
Comment (E-gy) [inline src/extercom/mod.rs:23]: This is a terrible pattern. You are not letting Rust compiler do the request-body-matches-rq-spec checks for you.

**Review by E-gy — CHANGES_REQUESTED** [review]
Single request for all is not the greatest idea, but at the same time, eh - time...

**Comment by E-gy** [comment]
> how do you add emojis?

Rocket has 1st class emoji support in routes! So, the `/fs` route can be `/🗄` or `/🗃` :D

**Comment by AristideEU** [comment]
> Single request for all is not the greatest idea, but at the same time, eh - time...

Yeah I kinda noticed when writing the "Message structure" and giving it so many optionnals, but i'd have to rewrite the client

File: src/extercom/mod.rs (hunk 2 above)
Comment (AristideEU) [reply src/extercom/mod.rs:23]: I have no Idea how it works, (heck i just learned rust yesterday) could you fix that please ?

File: src/extercom/mod.rs (hunk 1 above)
Comment (AristideEU) [reply src/extercom/mod.rs:10]: ctrrl-c ctrl-v

**Comment by E-gy** [comment]
The one in 🐍? Why?

**Comment by E-gy** [comment]
We still need the response types tho (with `-> Json<...>`, and put `unimplemented!()` in the body for now)

**Pull request merged**